Sources/SwiftProtobuf/JSONScanner.swift

```diff
@@ -706,6 +398,311 @@ internal struct JSONScanner {
     return Character(UnicodeScalar(UInt32(currentByte))!)
   }
 
+  // Parse the leading UInt64 from the provided utf8 bytes.
+  //
+  // This usually does a direct conversion of utf8 to UInt64.  It is
+  // called for both unquoted numbers and for numbers stored in quoted
+  // strings.  In the latter case, the caller is responsible for
+  // consuming the leading quote and verifying the trailing quote.
+  //
+  // If the number is in floating-point format, this uses a slower
+  // and less accurate approach: it identifies a substring comprising
+  // a float, and then uses Double() and UInt64() to convert that
+  // string to an unsigned intger.
+  //
+  // If it encounters a "\" backslash character, it returns a nil.  This
```

Comment:
So I had lots of comments at first about the backslash, because it wasn't until the final calling code that it made sense; so I'd suggest beefing up this comment.  The important detail that wasn't clear to me is this is called _first_ on the raw bytes in the JSON, so there is a chance a \uXXXX could be encountered when the double was quoted; so by returning nil, it is a signal that this couldn't directly parse the number and if it was a quoted string, that decode _must_ be done before and the result of that decode can then be tried for parsing again.